JeongJaeHeon/app.js

```diff
@@ -44,11 +49,65 @@ app.post('/users', async (req, res, next) => {
             profile_image
         ) VALUES (?, ?, ?, ?);
         `,
-          [userName, userEmail, userPassword, profileImage]
+          [userName, userEmail, hashedPassword, profileImage]
     ) 
     res.status(201).json({ message : "userCreated"})
 });
 
+app.get('/login', async(req, res) => {
+    const { userEmail, userPassword} = req.body
+
+    const user = 
+        await mysqlDataSource.query(
+            `SELECT
+                u.id,
+                u.name,
+                u.email.
+                u.password
+              FROM users u 
+            WHERE u.email=?`, [userEmail]
+        )
+    const hashedPassword = user[0].password
+
+    const match = await bcrypt.compare(userPassword, hashedPassword)
+            if(match === true) {
+                const payLoad = { userEmail : userEmail }
+                const secretKey = process.env.JWT_TOKEN_SECRET_KEY
+                const jwtToken = jwt.sign(payLoad, secretKey, {algorithm:"HS256", expiresIn:"7d"})
+
+                res.status(200).json({ accsessToken : jwtToken })
+            } else {
+                res.status(401).json({ message : "Invalid User" })
+            }
+})
+
+const validateToken = async(req, res, next) => {
+    const token = req.headers.authorization
+    const decoded = jwt.verify(token, process.env.JWT_TOKEN_SECRET_KEY)
+    if(!true) {
+        req.user = decoded
+        next()
+    } else {
+        res.status(404).json({ message : "Invalid Access Token" })
+    }
+}
```

Comment:
예외처리 시 token이 없는 경우, 복호화 된 정보가 유효하지 않은 경우로 나눌 수 있을 것 같습니다!

```suggestion
const validateToken = async(req, res, next) => {
    const token = req.headers.authorization
    const decoded = jwt.verify(token, process.env.JWT_TOKEN_SECRET_KEY)
    
    if (!token) {
      res.status(400).json({ message: "REQUIRED_LOGIN" })
    }
    if (!decoded) {
      res.status(400).json({ message: "INVALIED_USER" })
    }
    
    req.user = decoded
    next()
}
```